src/sys/ptrace/linux.rs

```diff
@@ -402,6 +425,28 @@ pub fn step<T: Into<Option<Signal>>>(pid: Pid, sig: T) -> Result<()> {
     }
 }
 
+/// Move the stopped tracee process forward by a single step or stop at the next syscall
+/// as with `ptrace(PTRACE_SYSEMU_SINGLESTEP, ...)`
+///
+/// Advances the execution by a single step or until the next syscall.
+/// In case the tracee is stopped at a syscall, the syscall will not be executed.
+/// Optionally, the signal specified by `sig` is delivered to the tracee upon continuation.
+#[cfg(all(target_os = "linux", target_env = "gnu", any(target_arch = "x86", target_arch = "x86_64")))]
+pub fn step_or_sysemu<T: Into<Option<Signal>>>(pid: Pid, sig: T) -> Result<()> {
```

Comment:
What do you think about `fn sysemu_singlestep`?

Reply:
I renamed it to `sysemu_step` to be in line with the naming of the `step` method.